deployment/modules/gcp/cloudbuild/main.tf

```diff
@@ -116,28 +106,77 @@ resource "google_cloudbuild_trigger" "build_trigger" {
     step {
-      id         = "cloud_run_deploy"
-      name       = "gcr.io/google.com/cloudsdktool/cloud-sdk"
-      entrypoint = "gcloud"
-      args = [
-        "run",
-        "deploy",
-        "${var.docker_env}-static-ct",
-        "--image",
-        "${local.conformance_gcp_docker_image}:$SHORT_SHA",
-        "--region",
-        var.location
+      id     = "terraform_apply_conformance_ci"
+      name   = "alpine/terragrunt"
+      script = <<EOT
+        terragrunt --terragrunt-non-interactive --terragrunt-no-color apply -auto-approve -no-color 2>&1
+        terragrunt --terragrunt-no-color output --raw conformance_url -no-color > /workspace/conformance_url
+      EOT
+      dir    = "deployment/live/gcp/ci"
+      env = [
+        "GOOGLE_PROJECT=${var.project_id}",
+        "TF_IN_AUTOMATION=1",
+        "TF_INPUT=false",
+        "TF_VAR_project_id=${var.project_id}"
       ]
       wait_for = ["docker_push_conformance_gcp"]
     }
 
-    ## TODO: Apply the terragrunt configuration to create the CI environment.
+    ## Since the conformance infrastructure is not publicly accessible, we need to use 
+    ## bearer tokens for the test to access them.
+    ## This step creates those, and stores them for later use.
+    step {
+      id       = "bearer_token"
+      name     = "gcr.io/cloud-builders/gcloud"
+      script   = <<EOT
+        curl -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/${local.cloudbuild_service_account}/identity?audience=$(cat /workspace/conformance_url)" > /workspace/cb_identity
+      EOT
+      wait_for = ["terraform_apply_conformance_ci"]
+    }
+
+    ## Test against the conformance server.
+    ## TODO: Replace this with CT Hammer when it is ready.
+    step {
+      id       = "curl_test"
+      name     = "gcr.io/cloud-builders/gcloud"
+      script   = <<EOT
+        apt update && apt install jq -y
+        mkdir -p /tmp/httpschain
+        openssl genrsa -out /tmp/httpschain/cert.key 2048
+        openssl req -new -key /tmp/httpschain/cert.key -out /tmp/httpschain/cert.csr -config=testdata/fake-ca.cfg
+        openssl x509 -req -days 3650 -in /tmp/httpschain/cert.csr -CAkey testdata/fake-ca.privkey.pem -CA testdata/fake-ca.cert -passin pass:"gently" -outform pem -out /tmp/httpschain/chain.pem -provider legacy -provider default
+        cat testdata/fake-ca.cert >> /tmp/httpschain/chain.pem
+        cat /tmp/httpschain/chain.pem | jq --raw-input --slurp --compact-output 'split("\n-----END CERTIFICATE-----\n") | map(select(length > 0) | sub("^-----BEGIN CERTIFICATE-----\n"; "") | sub("\n-----END CERTIFICATE-----$"; "")) | { "chain": . }' > /tmp/httpschain/chain.json
+        curl -i -X POST --data @/tmp/httpschain/chain.json -H "Content-Type: application/json" -H "Authorization: Bearer $(cat /workspace/cb_identity)" $(cat /workspace/conformance_url)/ci-${var.project_id}/ct/v1/add-chain
```

Comment:
Maybe get the output for curl and fail if it's not what expected: a success, and an index of 0.
It's a bit of a rabbit hole, we can go further, wait for a checkpoint, etc. But at this point, I don't think it's worth it - let's just finish building a proper conformance binary.